### PR TITLE
TPC-X query plan tests

### PR DESCRIPTION
**Pull request by max-hoffman**
Added schemas, stats, query plans for:
- TPC-H
- TPC-DS
- IMDB join planning benchmark

Added `plangen` to auto-update the tests.

We cannot parse all of the TPC-DS query plans yet. I saw some ROLLUP and aggregation validation errors.

Excluding [data ops benchmark](https://duckdb.org/2023/04/14/h2oai.html) because the plans are not interesting.

### CODE REVIEW

File: enginetest/scriptgen/setup/helper.go [hunk 1]
```diff
@@ -15,24 +15,29 @@
...
+	TpchPlanSetup        [][]SetupScript
+	TpccPlanSetup        [][]SetupScript
+	TpcdsPlanSetup       [][]SetupScript
+	ImdbPlanSetup        [][]SetupScript
+	LoadDataSetup        [][]SetupScript
+	XySetup              [][]SetupScript
+	JoinsSetup           [][]SetupScript
 )
 
 func init() {
```
Comment (zachmu) [inline enginetest/scriptgen/setup/helper.go:43]: What's the point of this init block, why can't these be declared inline

**Review by zachmu — APPROVED** [review]
Good to have these here, just a few minor comments on generation code

File: enginetest/plangen/testdata/spec.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    path: enginetest/queries/query_plans.go
+  - name: IndexPlanTests
+    path: enginetest/queries/index_query_plans.go
+  - name: IntegrationPlanTests
+    path: enginetest/queries/integration_plans.go
+  - name: TpchPlanTests
+    path: enginetest/queries/tpch_plans.go
+  - name: TpccPlanTests
+    path: enginetest/queries/tpcc_plans.go
+#  - name: TpcdsPlanTests
```
[the remark this answers is not in the document]
Comment (max-hoffman) [reply enginetest/plangen/testdata/spec.yaml:12]: we can't parse them yet

File: enginetest/plangen/cmd/plangen/main.go [hunk 3]
```diff
@@ -0,0 +1,297 @@
...
+	switch cmd {
+	case "plan":
+
+	default:
+		flag.Usage()
+		exit(errUnrecognizedCommand)
+	}
+
+	var err error
+	switch cmd {
```
[the remark this answers is not in the document]
Comment (max-hoffman) [reply enginetest/plangen/cmd/plangen/main.go:71]: deleted

**Pull request merged**